examples/twitter-nlp/analyze_sentiment.py

```diff
@@ -124,30 +130,41 @@ def _display_results(batch, sentiments):
     "--total_tweets",
     type=int,
     default=None,
-    help="The total number of tweets to analyze from the tweets_file."
+    help="The total number of tweets to analyze from the tweets_file. "
     "Defaults to None which will run through all tweets contained in the file.",
 )
+@click.option(
+    "--engine",
+    type=click.Choice(["deepsparse", "onnxruntime"]),
+    default="deepsparse",
+    help="Inference engine to use. Default is deepsparse",
+)
 def analyze_tweets_sentiment(
-    model_path: str, tweets_file: str, batch_size: int, total_tweets: int
+    model_path: str, tweets_file: str, batch_size: int, total_tweets: int, engine: str
 ):
     """
     Analyze the sentiment of the tweets given in the tweets_file and
     print out the results.
     """
+    print("Loading the model for inference...")
     text_pipeline = Pipeline.create(
         task="text-classification",
         model_path=model_path,
         batch_size=batch_size,
+        engine_type=engine,
     )
     tweets = _load_tweets(tweets_file)
     tweets = _prep_data(tweets, total_tweets)
     tot_sentiments = []
+    times = []
 
     while True:
         batch = _batched_model_input(tweets, batch_size)
         if batch is None:
             break
+        start = time.time()
```

Comment:
Would recommend using time.perf_counter() here instead. It uses a much more recent epoch to increase floating point resolution on the nanosecond scale. Planning to replace all time.time() calls in deepsparse early this week

Reply:
Didn't know about this, I'm all for this. It does seem to be slower in a microbenchmark but not by much.
https://stackoverflow.com/a/71068984

Reply:
Interesting find, that's good to know